generator/src/main/java/com/scottlogic/deg/generator/restrictions/MatchesStandardStringRestrictions.java

```diff
@@ -0,0 +1,124 @@
+package com.scottlogic.deg.generator.restrictions;
+
+import com.scottlogic.deg.generator.constraints.atomic.StandardConstraintTypes;
+import com.scottlogic.deg.generator.generation.IsinStringGenerator;
+import com.scottlogic.deg.generator.generation.SedolStringGenerator;
+import com.scottlogic.deg.generator.generation.StringGenerator;
+
+/**
+ * Represents the restriction of a field to an `aValid` operator
+ * Holds the type of the value that is required and whether the field has been negated
+ */
+public class MatchesStandardStringRestrictions implements StringRestrictions{
+    private final StandardConstraintTypes type;
+    private final boolean negated;
+
+    public MatchesStandardStringRestrictions(StandardConstraintTypes type, boolean negated) {
+        this.type = type;
+        this.negated = negated;
+    }
+
+    @Override
+    public boolean match(String x) {
+        return createGenerator().match(x);
+    }
+
+    public StringGenerator createGenerator() {
+        StringGenerator generator = getStringGenerator();
+
+        return negated
+            ? generator.complement()
+            : generator;
+    }
+
+    private StringGenerator getStringGenerator() {
+        switch (type){
+            case ISIN:
+                return new IsinStringGenerator();
+            case SEDOL:
+                return new SedolStringGenerator();
+        }
+
+        throw new UnsupportedOperationException(String.format("Unable to create string generator for: %s", type));
+    }
+
+    /**
+     * Will combine/intersect another StringRestrictions within this instance
+     *
+     * Rules are:
+     * Return this instance within modification if the other restrictions matches the following:
+     *   1. It is an equivalent MatchesStandardStringRestrictions instance (all properties match)
+     *   2. It is a TextualRestrictions that has:
+     *     2.1. no regex restrictions of any kind
+     *     2.2. any present length restrictions do not impact the ability to create any value
+     *
+     * @param other The other restrictions to combine/intersect
+     * @return Either this instance (success) or a NoStringsPossibleStringRestrictions if the other restrictions could not be merged or intersected
+     */
+    @Override
+    public StringRestrictions intersect(StringRestrictions other) {
+        if (other instanceof TextualRestrictions){
+            TextualRestrictions textualRestrictions = (TextualRestrictions) other;
+            if (!wouldAffectValuesProduced(textualRestrictions)){
```

Comment:
I'd swap the negation in this statement for readability. If "wouldAffectValues" return "noStrings"; otherwise return this

Reply:
Done